server/api/messageCenter/getUnreadMessages.ts

```diff
@@ -0,0 +1,34 @@
+import { Request, Response } from 'express'
+
+import { CountryIso } from '@meta/area'
+import { AssessmentName } from '@meta/assessment'
+import { MessageTopicType } from '@meta/messageCenter'
+
+import { AssessmentController } from '@server/controller/assessment'
+import { MessageCenterController } from '@server/controller/messageCenter'
+import Requests from '@server/utils/requests'
+
+export const getUnreadMessages = async (req: Request, res: Response) => {
+  try {
+    const { countryIso, assessmentName, cycleName, type } = req.query as {
+      countryIso: CountryIso
+      assessmentName: AssessmentName
+      cycleName: string
+      type: MessageTopicType
```

Comment:
You need to pass the topic key and the response should return only one unread count ( only for the combination of assessment, cycle , countryIso , topicKey)